client/landing/stepper/declarative-flow/internals/steps-repository/sensei-launch/use-atomic-site-plugins.tsx

```diff
@@ -0,0 +1,67 @@
+import { useDispatch } from '@wordpress/data';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { useDispatch as useRootDispatch, useSelector } from 'react-redux';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { SITE_STORE } from 'calypso/landing/stepper/stores';
+import {
+	fetchSitePlugins,
+	installPlugin as installPluginAction,
+} from 'calypso/state/plugins/installed/actions';
+import { getPlugins } from 'calypso/state/plugins/installed/selectors';
+import { Plugin } from '../sensei-purpose/purposes';
+
+interface InstalledPlugin {
+	id: string;
+	slug: string;
+	active: boolean;
+}
+
+export function useAtomicSitePlugins() {
```

Comment:
There is some room for improvement here but I think that we should do it as part of another PR if we get to it:
- I think that the flow of the method (wait for Sensei Pro to be installed, then queue the other plugins and wait for them to be installed) is not clear. It would clearer if this method handled polling and doing the steps one after the other.
- The SenseiLaunch component would just get the progress from this one (potentially changing the text when a new plugin is installed)

Reply:
Yeah, I wasn't entirely happy with it either. When we add code here to configure the atomic site after its up (eg saving the purpose option), we can improve it — especially because that'd also need to work with the progress reporting.